lib/ace/ext/papeeria/tex_completer.coffee

```diff
@@ -149,12 +166,45 @@ define((require, exports, module) ->
       else
         callback(null, @cache)
 
+  ###
+  # Show popup if token.type at current pos is one of allowedTypes[i]
+  # @ (editor) --> editor
+  # @ (list of strings) -- allowedTypes
+  ###
+  showPopupIfTokenIsOneOfTypes = (editor, allowedTypes) ->
+    if editor.completer?
+      pos = editor.getCursorPosition()
+      session = editor.getSession()
+      token = editor.session.getTokenAt(pos.row, pos.column)
+
+      if token?
+        for type in allowedTypes
+          if LatexParsingContext.isType(token, type)
+            editor.completer.showPopup(editor)
+            break
+
   class TexCompleter
       constructor: ->
-        @refGetter = new ReferenceGetter()
-      @init: (editor) ->  init(editor,  {win: "enter", mac: "enter"})
-
+        @refCache = new CompletionsCache(processReferenceJson)
+        @citeCache = new CompletionsCache(processCitationJson)
+      @init: (editor) ->
+        init(editor,  {win: "enter", mac: "enter"})
+
+        # we need two event handler becouse handler below work fine as well when we in
+        # existing \ref{|} or \cite{|}
+        # But doesn't work correct when we haven't ready block yet
```

Comment:
Come on, stop puzzling the reader. It doesn't work when you just've added `\ref{}` using autocomplete.